shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -280,6 +282,54 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
   flutterArguments.dart_entrypoint_argc = dartEntrypointArgs.size();
   flutterArguments.dart_entrypoint_argv = dartEntrypointArgs.data();
 
+  // Only create a Compositor if we have a ViewController.
+  if (_viewController) {
+    [_mainOpenGLContext makeCurrentContext];
+
+    auto interface = GrGLMakeNativeInterface();
+    GrContextOptions options;
+    auto context = GrDirectContext::MakeGL(GrGLMakeNativeInterface(), options);
```

Comment:
Move the GL stuff to `FlutterMacOSCompositor`, and call it `FlutterMacOSGLCompositor`.